torch/_tensor_str.py

```diff
@@ -80,8 +81,18 @@ def __init__(self, tensor):
             tensor_view = tensor.reshape(-1)
 
         if not self.floating_dtype:
+            self.complex_with_decimal = False
+            if self.complex_dtype:
+                # max width for complex tensors depends on whether or not tensor contains ints only
+                self.complex_with_decimal = sum([not (value.item().real.is_integer() and value.item().imag.is_integer())
+                                                for value in tensor_view])
             for value in tensor_view:
-                value_str = '{}'.format(value)
+                if self.complex_dtype and self.complex_with_decimal:
```

Comment:
```
if self.complex_dtype 
   if self.complex_with_decimal:
        value_str = ('{{:.{}f}}').format(PRINT_OPTS.precision).format(value)
   else:
        value_str = "{:.0f}".format(value.item())
```

Reply:
attribute name is okay though?

Reply:
I think we should change it to [this](https://github.com/pytorch/pytorch/pull/36331#discussion_r406515100) :D